### ISSUE TITLE
Android NDK: ERROR

### ISSUE THREAD

**Issue by handsetvn** [Bilibili/ijkplayer#2290]
/home/longtran/Desktop/ijkplayer-android/android
profiler build: NO
Android NDK: ERROR:/home/longtran/Desktop/ijkplayer-android/android/ijkplayer/ijkplayer-x86_64/src/main/jni/ffmpeg/Android.mk:ijkffmpeg: LOCAL_SRC_FILES points to a missing file    
Android NDK: Check that /home/longtran/Desktop/ijkplayer-android/android/contrib/build/ffmpeg-x86_64/output/libijkffmpeg.so exists  or that its path is correct   
/home/longtran/working/tool/android-ndk-r10e/build/core/prebuilt-library.mk:45: *** Android NDK: Aborting    .  Stop.
/home/longtran/Desktop/ijkplayer-android/android

**Comment by migueltarga** [comment]
````
./init-android.sh

cd android/contrib
./compile-ffmpeg.sh clean
./compile-ffmpeg.sh all

cd ..
./compile-ijk.sh all
```

**Comment by SepBreUp** [comment]
wzf@wzf-Aspire-4750:~/ijkplayer/android/contrib$ ./compile-ffmpeg.sh all
====================
[*] check archs
====================
FF_ALL_ARCHS = armv5 armv7a arm64 x86 x86_64
FF_ACT_ARCHS = armv5 armv7a arm64 x86 x86_64

====================
[*] check env armv5
====================
FF_ARCH=armv5
FF_BUILD_OPT=

--------------------
[*] make NDK standalone toolchain
--------------------
build on Linux x86_64
ANDROID_NDK=/home/wzf/android-ndk-r11b
IJK_NDK_REL=
You need the NDKr10e or later

**Comment by migueltarga** [comment]
@SepBreUp You need the NDKr10e or later

**Comment by SepBreUp** [comment]
@midoreigh I had used ndk,but without sdk ,I have to download sdk?

**Comment by midoreigh** [comment]
@SepBreUp Yes you will need sdk. If you can download and setup android studio, it will be great

**Comment by LSDsl** [comment]
Hi. I have 15.2 NDK but still see "You need the NDKr10e or later".

build on Linux x86_64
ANDROID_NDK=/home/lsdsl/Android/Sdk/ndk-bundle/
IJK_NDK_REL=15.2.4203891
You need the NDKr10e or later

**Comment by migueltarga** [comment]
@LSDsl add 15 to this line: https://github.com/Bilibili/ijkplayer/blob/master/android/contrib/tools/do-detect-env.sh#L71
and make a PR

**Comment by LSDsl** [comment]
ok. now ERROR: Failed to create toolchain.

build on Linux x86_64
ANDROID_NDK=/home/lsdsl/Android/Sdk/ndk-bundle/
IJK_NDK_REL=15.2.4203891
NDKr15.2.4203891 detected
HOST_OS=linux
HOST_EXE=
HOST_ARCH=x86_64
HOST_TAG=linux-x86_64
HOST_NUM_CPUS=1
BUILD_NUM_CPUS=2
Auto-config: --arch=arm
ERROR: Failed to create toolchain.

**Comment by migueltarga** [comment]
@LSDsl Try to use an older NDK for now.

**Comment by LSDsl** [comment]
Cannot select version in SDK manager :(

**Comment by diguagege** [comment]
You should modify do-detect-env.sh. 
case "$IJK_NDK_REL" in
            11*|12*|13*|14*|15*|16*|17*)
Like this

**Comment by paramvir-singh88** [comment]
@LSDsl did you find any solution for your issue? I am also getting the error "failed to create toolchain"

**Comment by fukemy** [comment]
no one can solve this

**Comment by testsharing** [comment]
I just used NDK r10e

**Comment by LSDsl** [comment]
I just used NDK r10e

**Comment by fukemy** [comment]
@LSDsl it work. but i can not build openssl mode

**Comment by LSDsl** [comment]
@fukemy And what is problem?

**Comment by JaioSkura** [comment]
Any solution for this? Without having to use NDK r10e?

**Comment by fukemy** [comment]
@LSDsl thanks. I  solved it, just used directly other implement library

**Comment by fukemy** [comment]
@LSDsl can you help? I got this error
not working for me, mac-os 64:
```
build on Darwin x86_64
ANDROID_NDK=/Users/thehe/Library/Android/sdk/ndk-bundle
NDKr10e(64-bit) detected
```

```
--------------------
# to continue to build ffmpeg, run script below,
sh compile-ffmpeg.sh 
# to continue to build ijkplayer, run script below,
sh compile-ijk.sh 
THG:contrib thehe$ cd ..
THG:android thehe$ ./compile-ijk.sh 
profiler build: NO
Android NDK: ERROR:/Users/thehe/ijkplayer-android/android/ijkplayer/ijkplayer-armv7a/src/main/jni/ffmpeg/Android.mk:ijkffmpeg: LOCAL_SRC_FILES points to a missing file    
Android NDK: Check that /libijkffmpeg.so exists  or that its path is correct   
/Users/thehe/Library/Android/sdk/ndk-bundle/build/core/prebuilt-library.mk:45: *** Android NDK: Aborting    .  Stop.
```

**Comment by LSDsl** [comment]
@fukemy first need build ffmpeg
./compile-ffmpeg.sh 
and after this
./compile-ijk.sh

**Comment by fukemy** [comment]
tks. it' work, i just follow step in stackoverflow but it's do not have compile ffmpeg

**Comment by ins1der1211** [comment]
You just need to add your ndk version to android/contrib/tools/do-detect-env.sh.
`case "$IJK_NDK_REL" in
            11*|12*|13*|14*|YOUR_NDK_VERSION*)
                if test -d ${ANDROID_NDK}/toolchains/arm-linux-androideabi-4.9
                then
                    echo "NDKr$IJK_NDK_REL detected"
                else
                    echo "You need the NDKr10e or later"
                    exit 1
                fi
            ;;
            *)
                echo "You need the NDKr10e or later"
                exit 1
            ;;
        esac`

**Comment by khyatiChitroda** [comment]
same issue I have, please help me,

profiler build: NO
Android NDK: ERROR:/home/khyati/ijkopenssl/ijkplayer/android/ijkplayer/ijkplayer-armv7a/src/main/jni/ffmpeg/Android.mk:ijkffmpeg: LOCAL_SRC_FILES points to a missing file
Android NDK: Check that /libijkffmpeg.so exists or that its path is correct
/home/khyati/Toolchain/android-ndk-r10e/build/core/prebuilt-library.mk:45: *** Android NDK: Aborting . Stop.
/home/khyati/ijkopenssl/ijkplayer/android

**Comment by 865394064** [comment]
you not ndk-r14e/. to build or ./compile-ffmpeg.sh all not success

**Comment by sachinklocham** [comment]
i was trying to use => android-ndk-r14b-linux-x86_64  but getting error "Failed to create toolchain."

my current python version on my system was Python 3.10.12 then
i changed it to Python 2.7.18 and my problem was solved.
